### ISSUE TITLE
Null Pointer Dereferencing in ae.h、dict.c、async.c、ivykis.h and libevent.h

### ISSUE THREAD

**Issue by salmonx** [redis/hiredis#751]
The following code has null pointers dereferencing, typically causing a crash or exit.

ae.h:111
```
    e = (redisAeEvents*)malloc(sizeof(*e));  // e may be null
    e->context = ac;
    e->loop = loop;
    e->fd = c->fd;
    e->reading = e->writing = 0;
```
dict.c:100
```
    n.table = calloc(realsize,sizeof(dictEntry*)); // n.table may be null
    ...
    for (i = 0; i < ht->size && ht->used > 0; i++) {
        if (ht->table[i] == NULL) continue;
        ....
        he = ht->table[i];
        ...
```
async.c:757
```
    if (!ac->c.timeout) {
        ac->c.timeout = calloc(1, sizeof(tv)); //ac->c.timeout  may be null
    }

    if (tv.tv_sec == ac->c.timeout->tv_sec &&
        tv.tv_usec == ac->c.timeout->tv_usec) {
        return;
    }
```
ivykis.h:58
```
    e = (redisIvykisEvents*)malloc(sizeof(*e)); // e may be null
    e->context = ac;
```
libevent.h:155
```
    e = (redisLibeventEvents*)calloc(1, sizeof(*e)); // e may be null
    e->context = ac;
```

**Comment by salmonx** [comment]
CVE-2020-7105 (https://security-tracker.debian.org/tracker/CVE-2020-7105)

**Comment by michael-grunder** [comment]
Fixed via #754
